File: fd-player/src/examples/ghost_bicyle_js/lib/wheel.js
// wheel([0, 0], 10, 2, 0)
return (center, outerRadius, innerRadius, angle) => {
  const spokeCount = 12;
  const baseAngle = angle ?? 0;
  const lines = Array.from({ length: spokeCount }, (_, index) => {
    const theta = baseAngle + index * ((Math.PI * 2) / spokeCount);
    return {
      type: 'line',
      data: {
        from: [
          center[0] + Math.sin(theta) * innerRadius,
          center[1] + Math.cos(theta) * innerRadius
        ],
        to: [
          center[0] + Math.sin(theta) * outerRadius,
          center[1] + Math.cos(theta) * outerRadius
        ],
        stroke: '#38bdf8',
        width: 0.3
      }
    };
  });
  return [
    lines,
    {
      type: 'circle',
      data: {
        center,
        radius: outerRadius,
        stroke: '#334155',
        width: 1,
        fill: null
      }
    },
    {
      type: 'circle',
      data: {
        center,
        radius: innerRadius,
        stroke: '#38bdf8',
        width: 1,
        fill: null
      }
    }
  ];
};

File: examples/testlib/art/cartoon/machine/parts/wheel.fs
// wheel([0, 0], 10, 2, 0)
(center, outerRadius, innerRadius, angle) =>
{
  n: 12;
  lines:
    Range(0, n) map (k, idx) =>
    {
      theta: (angle ?? 0) + k * (2 * math.Pi / n);
      eval
      {
        type: "line";
        from:
        [
          center[0] + math.Sin(theta) * innerRadius,
          center[1] + math.Cos(theta) * innerRadius
        ];
        to:
        [
          center[0] + math.Sin(theta) * outerRadius,
          center[1] + math.Cos(theta) * outerRadius
        ];
        stroke: "#38bdf8";
        width: 0.3;
      };
    };

  eval
    lines
    + [{ type: "circle"; center; radius: outerRadius; stroke: "#334155"; width: 1; fill: "none"; }]
    + [{ type: "circle"; center; radius: innerRadius; stroke: "#38bdf8"; width: 1; fill: "none"; }];
}
